source/Application/Model/User.php

```diff
@@ -1127,7 +1128,7 @@ public function checkValues($sLogin, $sPassword, $sPassword2, $aInvAddress, $aDe
         $oInputValidator->checkEmail($this, $sLogin);
 
         // 3. password
-        $oInputValidator->checkPassword($this, $sPassword, $sPassword2, ((int) Registry::getConfig()->getRequestParameter('option') == 3));
+        $oInputValidator->checkPassword($sLogin, $sPassword, $sPassword2, ((int) Registry::getConfig()->getRequestParameter('option') == 3));
```

Comment:
We can not change what we pass to this method as first argument. We clearly do not use this, but others might chain overload the `InputValidtor` class and rely on the first parameter being the `User` model.
```suggestion
        $oInputValidator->checkPassword($this, $sPassword, $sPassword2, ((int) Registry::getConfig()->getRequestParameter('option') == 3));
```